libraries/AP_Airspeed/AP_Airspeed_Health.cpp

```diff
@@ -37,7 +37,9 @@ void AP_Airspeed::check_sensor_ahrs_wind_max_failures(uint8_t i)
     }
 
     const AP_GPS &gps = AP::gps();
-    if (gps.status() < AP_GPS::GPS_Status::GPS_OK_FIX_3D) {
+    constexpr uint8_t gps_instance = 0;
+    const auto gps_status = gps.status(gps_instance);
+    if (gps_status < AP_GPS::GPS_Status::GPS_OK_FIX_2D) {
```

Comment:
Not a fan, Firstly your moving to use the first instance rather than the primary, in some cases this will be much worse. For example some (incorrectly) have a second GPS but not a first one. 

Secondly I suspect if you look at GPS velocity for just a 2D fix it will be very bad.

Reply:
Ah, this should be the primary instance, not zero. Will fix. 

Rather than qualifying based on the GPS Fix status, what about using the ratio of `AP_GPS.speed_accuracy` to  `AHRS_WIND_MAX`?

Reply:
The primary sensor is now explicitly used in 8c6a36fea615b1e933579450269a06a9fd8b29f2. 

In the future, perhaps this class could rely on the fused GPS output.

Reply:
I guess speed accuracy might work, however this code is not used very much, I think we would probably want evidence of the current approach going wrong before we change it. Or at least a log that shows this would not be worse.

Reply:
Would you accept SITL logs, or only from a real aircraft? I could do either, but it will take time. I'm in no rush to merge this, just trying to make steady improvements to the effectiveness of the failure detection and recovery.

Reply:
I think probably a real log, @tridge what do you think?

The 2D to 3D is a clear improvement, I think that part could go in right away.

Reply:
BTW, it doesn't necessarily mean you have to generate a new log, we already log the necessary info so we might be able to find a old log where you can show it would be better.

Reply:
> Not a fan, Firstly your moving to use the first instance rather than the primary, in some cases this will be much worse. For example some (incorrectly) have a second GPS but not a first one.
> 
> Secondly I suspect if you look at GPS velocity for just a 2D fix it will be very bad.

What I was trying to get at with the change here was more of this:
* What fix type do we want to use only ground speed (if at all)
* What fix type do we require using 3D velocity to compute speed

My intent was to support 2D fix for ground speed, but it obviously lowers the barrier to assuming GPS is better than airspeed. 

I'm happy to remove it, and remove the usage of ground speed, switching to the 3D math altogether. This would require we check GPS has a 3D fix before using the data.